alignment.py

```diff
@@ -0,0 +1,109 @@
+import cv2 as cv
+import numpy as np
+import math
+
+distance_for_match = 1.7976931348623157e308
+
+
+def read_key_points_from_file(path):
+    with open(path, "r") as f:
+        lines = [line.rstrip() for line in f]
+    key_points = list()
+    for line in lines:
+        point = list(map(int, line.split()))
+        key_points.append(cv.KeyPoint(point[0], point[1], 1))
+    return key_points
+
+
+def calculate_point_in_destination_image(image_source, pixel, homography_matrix):
+    height, width = image_source.shape[:2]
+    if pixel.pt[0] < 0 or pixel.pt[0] > width:
+        return
+    if pixel.pt[1] < 0 or pixel.pt[1] > height:
+        return
+    point_in_destination_image = np.array(homography_matrix).dot(
```

Comment:
Let's use `perspectiveTransform` from OpenCV ([example](https://stackoverflow.com/a/55055829)).
It seems that there is an error in your calculations now, you can check [here](https://stackoverflow.com/a/9275631).